src/errors/types.rs

```diff
@@ -151,9 +151,9 @@ pub enum ErrorType {
     // dict errors
     #[strum(message = "Input should be a valid dictionary")]
     DictType,
-    #[strum(message = "Unable to convert mapping to a dictionary, error: {error}")]
-    DictFromMapping {
-        error: String,
+    #[strum(message = "Input should be a valid mapping, error: {error}")]
+    MappingType {
+        error: Cow<'static, str>,
```

Comment:
@samuelcolvin What's the downside to use `String` type here? Since it still passed the CI.

Reply:
In the case where we use a `&'static str` error, we can avoid allocating a new string, and just use the reference to that static string. But this also allows us to use a `String` if we want something else.

`Cow` here just means

```rs
enum Cow {
    Borrowed(&'static str),
    Owned(String)
}
```

Reply:
Thanks, that makes sense to me.